### ISSUE TITLE
SDL does not send GENERIC_ERROR in case HMI does NOT respond during <DefaultTimeout>+<RPCs_internal_timeout>

### ISSUE THREAD

**Issue by LitvinenkoIra** [smartdevicelink/sdl_core#1880]
**Occurrence: Always** 

# Description
SDL must send GENERIC_ERROR to app in case HMI does NOT respond during DefaultTimeout+2*RPCs_internal_timeout for all RPCs with own timer

## Preconditions
- Core, HMI started.
- App is registered on HMI.
- App is activated.

## Steps to reproduce
1.  SDL transfers *RPC with own timeout from mobile app to HMI 
and HMI does NOT respond during DefaultTimeout+2*RPCs_own_timeout

Script for reproducing: https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/develop/test_scripts/Defects/4_5/1880_DefaultAndRPCTimeoutGENERIC_ERROR.lua

## Actual result
N/A

## Expected result
SDL must respond 'GENERIC_ERROR, success:false' to mobile app

# Environment

# Attachments
Note: Extract attached folder with script to test_scripts folder. 
[Defects.zip](https://github.com/smartdevicelink/sdl_core/files/1512868/Defects.zip)

# Expected delivery
- [ ] Source code updates
- [ ] Code comments
- [x] **UTs add/update _(not required)_**
- [ ] ATF tests add/update
- [x] **Manual tests _(not required)_**
- [x] **Add/update CI plans/jobs _(not required)_**
- [x] **SDD updates _(not required)_**
- [x] **Guidelines update ([sdl_core_guides](https://github.com/smartdevicelink/sdl_core_guides)) _(not required)_**
- [x] **Guidelines update ([sdl_hmi_integration_guidelines](https://github.com/smartdevicelink/sdl_hmi_integration_guidelines)) _(not required)_**

**Comment by aderiabin** [comment]
Contributor priority is set `High` with reason: The issue is related to critical functionality: existence of response from HMI/SDL.

**Comment by JackLivio** [comment]
Similar issue to #1871 and #792

**Comment by jacobkeeler** [comment]
Looks like a duplicate of #792, but with the specification of `2*RPCs_internal_timeout` added to the default timeout instead. Why is this?

**Comment by yang1070** [comment]
@LitvinenkoIra I have the same question as Jacob here, Why is 2*RPCs_internal_timeout + default?

**Comment by GetmanetsIrina** [comment]
@JackLivio , @yang1070 , according to requirements:
SDL must start `DefaultTimeout(from .ini file) + RPCs_internal_timeout` for waiting the response from HMI side.
_RPCs_internal_timeout for RPCs:_
a. Alert (“duration” param)
b. PerformInteraction (“timeout” param -> increased two times for BOTH and MANUAL_ONLY modes)
c. Slider (“timeout” param)
d. ScrollableMessage (“timeout” param)

So described case `DefaultTimeout+2*RPCs_internal_timeout` is related to  `PerformInteraction` RPC with `BOTH` and `MANUAL_ONLY` modes

**Comment by yang1070** [comment]
@GetmanetsIrina 
I'm kind of understand the requirement in the case of `PerformInteraction` RPC with` BOTH` mode, as the work flow is to show VR portion, after some system defined time out  (handled by the platform), it shows the MAUNAL portion (the timeout paramter in RPC). So the sdl wait time shall be longer than DefaultTimeout+RPCs_internal_timeout due to the added VR portion. But it does not specify how much longer as we don't known what's the timeout for VR portion. This might be the reason that we treat it as the same as the timeout parameter. Therefore the result DefaultTimeout+2*RPCs_internal_timeout.


But why`MANUAL_ONLY` mode also need DefaultTimeout+2*RPCs_internal_timeout?


From the mobile API, I get the following description in `PerformInteraction` RPC
```xml
        <param name="timeout" type="Integer" minvalue="5000" maxvalue="100000" defvalue="10000" mandatory="false">
            <description>
                Timeout in milliseconds.
                If omitted a standard value of 10000 milliseconds is used.
                Applies only to the menu portion of the interaction. The VR timeout will be handled by the platform.
            </description>
        </param>
```

and from integration guideline, I get the following description about timeout
```
Timeout in milliseconds. 
The amount of time the HMI must wait for the   User to make a choice. 
After the time is up HMI respond with timed_out result   code.
```

**Comment by GetmanetsIrina** [comment]
@yang1070 , @jacobkeeler , @JackLivio 
According to discussion in emails the expected behavior is the same, updated with more details:

### Steps to reproduce
1. SDL transfers RPC with own timeout from mobile app to HMI _with the same timeout value received from mobile app_
    and HMI does NOT respond during DefaultTimeout+*RPCs_own_timeout

### Note:
RPCs_own_timeout for RPCs:
a. Alert (“duration” param)
b. PerformInteraction (“timeout” param -> increased two times for BOTH and MANUAL_ONLY modes)
c. Slider (“timeout” param)
d. ScrollableMessage (“timeout” param)

**Comment by JackLivio** [comment]
Fixed by #2495
